docs/io/index_unofficial.rst

```diff
@@ -0,0 +1,283 @@
+  `ctapipe.io.eventfilereader.EventFileReader` in
+  `ctapipe.io.unofficial.eventfilereader` to ensure the common high-level
+  interface for reading data is maintained.
+* Care should be taken in the imports of external modules. There should be no
+  dependencies added to ctapipe on the external libraries required for these
+  unofficial readers.
+
+
+How to use my reader in ctapipe?
+================================
+
+Presumably you already have some software that exists to read in your data
+format. What form that reader exists in dictates how easy it is to incorporate
+into ctapipe. Some of the possible forms and appropriate inplementations are:
+
+* **Your reader can be entirely programmed in Python**, in which case your
+  reader does not depend on any external packages, and you can include that
+  code in your reader's :ref:`Low-Level Reader-Specific Modules`.
+* **Your reader exists as an external Python module**, in which case it could
+  be made available on conda for easy (optional) installion.
+* **You have a C/C++ inplementation of your reader in a simple script**,
+  therefore you could provide an interface to that functionality by using
+  "ctypes". The C/C++ scripts can exist inside your reader module, and are
+  compiled as part of the ctapipe installation. See
+  `ctapipe.utils.neighbour_sum` as an example of how to interface Python with
+  C code, and remember to add your extension to ext_modules in setup.py.
+* **Your reader exists as an external C/C++ library**, therefore software such
+  as SWIG can be used to provide a Python wrapper to the library,
+  automatically creating interfaces to the functions in the library.
+
+
+EventFileReaders (`ctapipe.io.unofficial.eventfilereader`)
+==========================================================
+
+This class provides the high-level interface for reading data from each data
+format. Read about the purpose of EventFileReaders at
+:ref:`EventFileReaderClasses`.
+
+Location for every unofficial `ctapipe.io.eventfilereader.EventFileReader`.
+
+Each reader is required to import their
+:ref:`Low-Level Reader-Specific Modules` inside their `__init__` method,
+therefore bypassing the need for the external reader software to be installed
+unless someone has chosen to use this specific
+`ctapipe.io.eventfilereader.EventFileReader`, in which case a
+`ModuleNotFoundError` will be raised if the external software has not been
+found.
+
+The reader must also implement a fast method for
+`ctapipe.io.eventfilereader.EventFileReader.check_file_compatibility`. This
+method attempts to find a reader which *may* be compatible with the input
+file, and can be as simple as looking for a unique file extension in the file
+path or attempting to find a header keyword in the file. It is strongly
+advised that this method does not depend on the external reader software, as
+this would introduce a dependency when choosing the correct reader in the
+`ctapipe.io.eventfilereader.EventFileReaderFactory`.
```

Comment:
I think we are mixing topics here, or features. As we have discussed already, using a reader within that factory makes sense only if that reader is able to provide containers which are expected by the user of the factory.
But at this point this important fact has not even been mentioned.

Therefore, I propose:

 * rename `EventFileReader.check_file_compatibility(path)` to `EventFileReader.is_compatible(path)`
 * make the abstract method `is_compatible` explicit and by default return `False`.

This way, all new readers, whose developers do not yet know about that factory, simply work. The factory will just ignore them. Until the developers read further about that factory.

Reply:
If the creator of the Reader has correctly followed these guidelines and the existing Reader implementations, then their Reader **will** provide containers which are expected in the generic scripts which utilise EventFileReaderFactory to process waveforms without caring about the data format they come from. Such as examples/calibration_pipeline.py.

Allowing the creator to ignore EventFileReaderFactory and check_file_compatibility will cause confusion when trying to read their data format into this script, as they will be told that no EventFileReader is compatible.

I think its fine to expect the creator to setup the full implementation of the EventFileReader.

Reply:
The only containers that could be data format specific are the R0 containers. But they should still contain the same fields as R0CameraContainer (adc_samples, adc_sums), so the common waveform processing scripts will always find the correct fields irregardless of data format, so their should be no problem with using the factory.

Reply:
There are camera types, which do not sample waveforms, right?

Reply:
Sure, but they are treated as waveforms with 1 sample, and are handled specially when needed (e.g. in the dl1 calibration no processing is done to the values).